src/utils/ga.js

```diff
@@ -1,29 +1,54 @@
-/* @START-EXCLUDE: 'lib' */
-import ReactGA from 'react-ga';
-/* @END-EXCLUDE */
+export const LogCategories = Object.freeze({
+    ChartTitle:  'Chart Title',
+    CategoricalDisplay:  'Categorical Display',
+    ChartControl: 'Chart Control',
+});
 
+export const LogActions = Object.freeze({
+    MarketSelector: 'Market Selector',
+    Favorite: 'Favorite',
+    ChartSetting: 'Chart Setting',
+    ChartSize: 'Chart Size',
+    ChartType: 'Chart Type',
+    Comparison: 'Comparison',
+    DrawTools: 'Draw Tools',
+    Download: 'Download',
+    Indicator: 'Indicator',
+    Interval: 'Interval',
+    Template: 'Template',
+});
+
+/* eslint-disable */
 export function initGA() {
     /* @START-EXCLUDE: 'lib' */
-    if (!window.isProductionWebsite()) return;
-    ReactGA.initialize('UA-40877026-15');
+    if (!window.isProductionWebsite) return;
+    (function (i, s, o, g, r, a, m) {
+        i.GoogleAnalyticsObject = r; i[r] = i[r] || function () {
+            (i[r].q = i[r].q || []).push(arguments);
+        }, i[r].l = 1 * new Date(); a = s.createElement(o),
+        m = s.getElementsByTagName(o)
+            [0]; a.async = 1; a.src = g; m.parentNode.insertBefore(a, m);
+    }(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga'));
+    ga('create', 'UA-40877026-15', 'auto');
     /* @END-EXCLUDE */
 }
 
 export function logPageView() {
     /* @START-EXCLUDE: 'lib' */
-    if (!window.isProductionWebsite()) return;
-    ReactGA.set({ page : window.location.pathname });
-    ReactGA.pageview(window.location.pathname);
+    if (!window.isProductionWebsite) return;
+    try {
+        ga('send', 'pageview', window.location.pathname);
+    }
+    catch{}
     /* @END-EXCLUDE */
 }
 
 export function logEvent(category, action, label) {
     /* @START-EXCLUDE: 'lib' */
-    if (!window.isProductionWebsite()) return;
-    ReactGA.event({
-        category,
-        action,
-        label,
-    });
+    if (!window.isProductionWebsite) return;
+    try {
```

Comment:
instead of a try catch block you can set `ga` implementation to be just a blank function.

Reply:
@brucebinary its just a precaution, i imported analytics.js before ga is being used. but since its being imported asynchronously, i add try catch. why do we need a back function?